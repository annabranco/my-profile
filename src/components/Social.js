import React from 'react';

class Social extends React.Component {

	render () {

		const text = this.props.text;
		const language = this.props.language;

		return (

			<div className="hero__social">
				<div className="hero__social--inner">
					<a className="hero__social--link-mail" href="mailto:anya.branco@icloud.com">
						<i className="far fa-envelope icon-mail"></i>
						<p className="hero__social-text">anya.branco@icloud.com</p>
					</a>
				</div>
				<div className="hero__social--inner">
					<a className="hero__social--link" href="https://github.com/annabranco" target="_Blank" rel="noopener noreferrer">
						<i className="fab fa-github-alt icon-github"></i>
						<p className="hero__social-text">Github</p>
					</a>
				</div>
				<div className="hero__social--inner">
					<a className="hero__social--link" href="https://www.linkedin.com/in/annabranco/" target="_Blank" rel="noopener noreferrer">
						<i className="fab fa-linkedin-in icon-linkedin"></i>
						<p className="hero__social-text">LinkedIn</p>
				</a>
				</div>
				<div className="hero__social--inner">
					<a className="hero__social--link" href="https://twitter.com/AnyaBranco" target="_Blank" rel="noopener noreferrer">
						<i className="fab fa-twitter icon-twitter"></i>
						<p className="hero__social-text">Twitter</p>
				</a>
				</div>
				<div className="hero__social--inner hero__social--link">
					<a className="hero__social--link hero__social--link-skype" href="skype:live%3Aanna.branco_3?call">
						<i className="fab fa-skype icon-call"></i>
						<p className="hero__social-text">{text[language].call}</p>
					</a>
					<a className="hero__social--link hero__social--link-skype" href="skype:live%3Aanna.branco_3?chat">
						<i className="fas fa-comments icon-chat"></i>
						<p className="hero__social-text">{text[language].chat}</p>
					</a>
				</div>
			</div>

		);
	}

}

export default Social;
