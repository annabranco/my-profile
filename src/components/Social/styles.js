import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueDark,
  colorBlack,
  fontTitleAlt,
  colorGreenBright,
  colorBlueNormal
} from '../../styles/theme';

export const Icon = styled.div`
  display: inline;
  vertical-align: middle;
  text-shadow: 0 0 1px ${colorBlack};
  font-size: 2rem;
`;
Icon.displayName = 'Icon';

export const Link = styled.a`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  text-decoration: none;

  color: ${colorBlueNormal};
  & :hover {
    color: ${colorBlueDark};
  }
  &:hover {
    text-decoration: none;
  }
`;
Link.displayName = 'Link';

export const LinkButton = styled(Link)`
  border: 1px solid ${colorWhite};
  border-radius: 25px 0;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorGreenBright, 0.5)}
  );
  width: 120px;
  padding: 5px 0;
  text-decoration: none;
  box-shadow: 0 4px 5px 0 ${rgba(colorBlack, 0.4)},
    inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};

  @media all and (min-width: 321px) {
    width: 150px;
  }

  @media all and (min-width: 500px) {
    width: 200px;
  }

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px ${rgba(colorBlack, 0.4)};
  }
`;
LinkButton.displayName = 'Link Button';

export const LinkSkypeButton = styled(LinkButton)`
  width: 75px;
  flex-direction: column;
  width: 100%;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }

  @media all and (min-width: 500px) {
    width: 50%;
  }

  &:first-of-type {
    border-radius: 25px 0 0 0;
  }

  &:last-of-type {
    border-radius: 0 0 25px 0;
  }
`;
LinkSkypeButton.displayName = 'Link Button ';

export const SocialArea = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border-radius: 30px;
  background-image: linear-gradient(
    ${rgba(colorGreenBright, 0.1)},
    ${rgba(colorWhite, 0.2)}
  );
  height: 400px;
  padding: 10px 0 30px;

  @media all and (min-width: 500px) {
    height: 500px;
    padding: 10px 0 80px;
  }
`;
SocialArea.displayName = 'Social Area';

export const SocialItem = styled.div`
  margin: 5px 0;
  width: 200px;

  @media all and (min-width: 500px) {
    margin: 15px 0;
  }
`;
SocialItem.displayName = 'Social Item';

export const SocialItemSkype = styled(SocialItem)`
  display: flex;
  justify-content: center;
  align-items: center;
`;
SocialItemSkype.displayName = 'Social Item';

export const Text = styled.p`
  margin: 4px 0 0;
  font-family: ${fontTitleAlt};
  font-size: 1rem;
  color: ${colorBlack};

  ${LinkButton} & {
    margin: 4px 0 0 25px;
  }

  ${LinkSkypeButton} & {
    margin: 4px 0 0 0;
  }

  @media all and (min-width: 321px) {
    margin: 4px 0 0 15px;
  }

  @media all and (min-width: 500px) {
    margin: 4px 0 0 0;
    font-size: 1.3rem;
  }
`;
Text.displayName = 'Text';
