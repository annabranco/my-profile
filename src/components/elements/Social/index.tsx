import { ReactElement } from 'react';
import { IInfoPageText } from '../../../types/interfaces';
import { infoPageTextPropType } from '../../../types/propTypes';
import {
  Icon,
  Link,
  LinkButton,
  LinkSkypeButton,
  SocialArea,
  SocialItem,
  SocialItemSkype,
  Text
} from './styles';

const Social = ({ texts }: { texts: IInfoPageText }): ReactElement => (
  <SocialArea>
    <SocialItem>
      <Link data-e2e-id="social-email" href="mailto:anya.branco@icloud.com">
        <Icon className="far fa-envelope" />
        <Text>anya.branco@icloud.com</Text>
      </Link>
    </SocialItem>
    <SocialItem>
      <LinkButton
        data-e2e-id="social-github"
        href="https://github.com/annabranco"
        target="_Blank"
        rel="noopener noreferrer"
      >
        <Icon className="fab fa-github-alt" />
        <Text>Github</Text>
      </LinkButton>
    </SocialItem>
    <SocialItem>
      <LinkButton
        data-e2e-id="social-linkedin"
        href="https://www.linkedin.com/in/annabranco/"
        target="_Blank"
        rel="noopener noreferrer"
      >
        <Icon className="fab fa-linkedin-in" />
        <Text>LinkedIn</Text>
      </LinkButton>
    </SocialItem>
    <SocialItem>
      <LinkButton
        data-e2e-id="social-twitter"
        href="https://twitter.com/AnyaBranco"
        target="_Blank"
        rel="noopener noreferrer"
      >
        <Icon className="fab fa-twitter" />
        <Text>Twitter</Text>
      </LinkButton>
    </SocialItem>
    <SocialItemSkype>
      <LinkSkypeButton
        data-e2e-id="social-skype_call"
        href="skype:live%3Aanna.branco_3?call"
      >
        <Icon className="fab fa-skype" />
        <Text>{texts.call}</Text>
      </LinkSkypeButton>
      <LinkSkypeButton
        data-e2e-id="social-skype_chat"
        href="skype:live%3Aanna.branco_3?chat"
      >
        {' '}
        <Icon className="fas fa-comments" />
        <Text>{texts.chat}</Text>
      </LinkSkypeButton>
    </SocialItemSkype>
  </SocialArea>
);

Social.propTypes = {
  texts: infoPageTextPropType.isRequired
};

export default Social;
