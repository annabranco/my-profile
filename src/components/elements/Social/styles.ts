import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlack,
  colorBlueDark,
  colorBlueNormal,
  colorGreenBright,
  colorWhite,
  fontTitleAlt
} from '../../../styles/theme';

export const Icon = styled.div`
  display: inline;
  vertical-align: middle;
  text-shadow: 0 0 1px ${colorBlack};
  font-size: 2rem;
`;
Icon.displayName = 'Icon';

export const Link = styled.a`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  text-decoration: none;

  color: ${colorBlueNormal};
  & :hover {
    color: ${colorBlueDark};
  }
  &:hover {
    text-decoration: none;
  }
`;
Link.displayName = 'Link';

export const LinkButton = styled(Link)`
  border: 1px solid ${colorWhite};
  border-radius: 25px 0;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorGreenBright, 0.5)}
  );
  width: 120px;
  padding: 5px 0;
  text-decoration: none;
  box-shadow: 0 4px 5px 0 ${rgba(colorBlack, 0.4)},
    inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};

  @media all and (min-width: 321px) {
    width: 150px;
  }

  @media all and (min-width: 500px) {
    width: 200px;
  }

  @media all and (min-width: 2000px) {
    margin-left: -50px;
    height: 80px;
    width: 300px;
  }

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px ${rgba(colorBlack, 0.4)};
  }
`;
LinkButton.displayName = 'Link Button';

export const LinkSkypeButton = styled(LinkButton)`
  width: 75px;
  flex-direction: column;
  width: 100%;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }

  @media all and (min-width: 500px) {
    width: 50%;
  }

  @media all and (min-width: 2000px) {
    height: 80px;
    width: 150px;
  }

  &:first-of-type {
    border-radius: 25px 0 0 0;
  }

  &:last-of-type {
    border-radius: 0 0 25px 0;
  }
`;
LinkSkypeButton.displayName = 'Link Button ';

export const SocialArea = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border-radius: 30px;
  background-image: linear-gradient(
    ${rgba(colorGreenBright, 0.1)},
    ${rgba(colorWhite, 0.2)}
  );
  height: 100%;
  padding: 10px 0 30px;

  @media all and (min-width: 500px) {
    height: 120%;
    padding: 10px 0 80px;
  }
`;
SocialArea.displayName = 'Social Area';

export const SocialItem = styled.div`
  margin: 5px 0;
  width: 150px;

  @media all and (min-width: 500px) {
    margin: 15px 0;
    width: 200px;
  }
`;
SocialItem.displayName = 'Social Item';

export const SocialItemSkype = styled(SocialItem)`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 150px;

  @media all and (min-width: 768px) {
    width: 200px;
  }
  @media all and (min-width: 2000px) {
    margin-left: 50px;
    justify-content: space-between;
    width: 250px;
  }
`;
SocialItemSkype.displayName = 'Social Item';

export const Text = styled.p`
  margin: 4px 0 0;
  font-family: ${fontTitleAlt};
  font-size: 1rem;
  color: ${colorBlack};

  ${LinkButton} & {
    margin: 4px 0 0 25px;
  }

  ${LinkSkypeButton} & {
    margin: 4px 0 0 0;
  }

  @media all and (min-width: 321px) {
    margin: 4px 0 0 15px;
  }

  @media all and (min-width: 500px) {
    margin: 4px 0 0 0;
    font-size: 1.3rem;
  }
  @media all and (min-width: 2000px) {
    font-size: 2rem;
  }
`;
Text.displayName = 'Text';
