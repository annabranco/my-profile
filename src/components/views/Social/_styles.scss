.infoPage__social {
	border-radius   : 30px;
	background-image: linear-gradient(rgba($green-bright, 0.1),rgba($white, 0.2));
	height          : 400px;
	display         : flex;
	padding         : 10px 0 30px;
	flex-direction  : column;
	align-items     : center;
	justify-content : space-around;

	@media all and (min-width: 500px) {
		height : 500px;
		padding: 10px 0 80px;
	}
}

.infoPage__social--inner {
	margin: 5px 0;

	@media all and (min-width: 500px) {
		margin: 15px 0;
	}
}

.infoPage__social--link,
.infoPage__social--link-mail,
.infoPage__social--link-skype {
	display        : flex;
	align-items    : center;
	justify-content: center;
}

.infoPage__social--link {
	border          : 1px solid $white;
	border-radius   : 25px 0;
	background-image: linear-gradient(rgba($white, 0.2),rgba($green-bright, 0.5));
	width           : 120px;
	padding         : 5px 0;
	text-decoration : none;
	box-shadow      : 0 4px 5px 0 rgba($black, 0.4), inset 0 1px 5px 1px rgba($white, 0.7);

	&:hover {
		text-decoration: none;
		transform      : translate(0,1px);
		box-shadow     : inset 0 2px 5px 1px rgba($black, 0.4);
	}

	@media all and (min-width: 321px) {
		width: 150px;
	}

	@media all and (min-width: 500px) {
		width: 200px;
	}
}

.infoPage__social--inner + .infoPage__social--link {
	border          : 0;
	border-radius   : 0;
	background-image: none;
	width           : 200px;
	padding         : 0;
	text-decoration : none;
	box-shadow      : none;

	&:hover {
		text-decoration: none;
		transform      : none;
		box-shadow     : none;
	}
}

.infoPage__social--link-mail,
.infoPage__social--link-skype {
	flex-direction : column;
	width          : 100%;
	text-decoration: none;

	&:hover {
		text-decoration: none;
	}
}

.infoPage__social--link-skype {
	width: 75px;

	&:first-of-type {
		border-radius: 25px 0 0 0;
	}

	&:last-of-type {
		border-radius: 0 0 25px 0;
	}

	@media all and (min-width: 500px) {
		width: 50%;
	}
}

.icon-call,
.icon-chat,
.icon-github,
.icon-linkedin,
.icon-mail,
.icon-twitter {
	display       : inline;
	vertical-align: middle;
	font-size     : 2rem;
	text-shadow   : 0 0 1px $black;
}

.icon-call,
.icon-chat {
	.infoPage__social--link-skype:hover & {
		color: $blue--dark;
	}
}

.icon-github,
.icon-linkedin,
.icon-mail,
.icon-twitter {
	display       : inline;
	vertical-align: middle;
	font-size     : 2rem;
	text-shadow   : 0 0 1px $black;

	.infoPage__social--link:hover & {
		color: $blue--dark;
	}
}

.infoPage__social-text {
	margin     : 4px 0 0 5px;
	font-family: $font__title-alt;
	font-size  : 1rem;
	color      : $black;

	.infoPage__social--link-mail &,
	.infoPage__social--link-skype & {
		margin: 4px 0 0;
	}

	@media all and (min-width: 321px) {
		margin: 4px 0 0 15px;
	}

	@media all and (min-width: 500px) {
		margin   : 4px 0 0 25px;
		font-size: 1.3rem;

		.infoPage__social--link-mail &,
		.infoPage__social--link-skype & {
			margin: 4px 25px 0;
		}
	}
}
