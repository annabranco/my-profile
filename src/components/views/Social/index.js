import React from 'react';
import { infoPageTextPropType } from '../../../types';

const Social = ({ texts }) => (
  <div className="infoPage__social">
    <div className="infoPage__social--inner">
      <a
        className="infoPage__social--link-mail"
        href="mailto:anya.branco@icloud.com"
      >
        <i className="far fa-envelope icon-mail" />
        <p className="infoPage__social-text">anya.branco@icloud.com</p>
      </a>
    </div>
    <div className="infoPage__social--inner">
      <a
        className="infoPage__social--link"
        href="https://github.com/annabranco"
        target="_Blank"
        rel="noopener noreferrer"
      >
        <i className="fab fa-github-alt icon-github" />
        <p className="infoPage__social-text">Github</p>
      </a>
    </div>
    <div className="infoPage__social--inner">
      <a
        className="infoPage__social--link"
        href="https://www.linkedin.com/in/annabranco/"
        target="_Blank"
        rel="noopener noreferrer"
      >
        <i className="fab fa-linkedin-in icon-linkedin" />
        <p className="infoPage__social-text">LinkedIn</p>
      </a>
    </div>
    <div className="infoPage__social--inner">
      <a
        className="infoPage__social--link"
        href="https://twitter.com/AnyaBranco"
        target="_Blank"
        rel="noopener noreferrer"
      >
        <i className="fab fa-twitter icon-twitter" />
        <p className="infoPage__social-text">Twitter</p>
      </a>
    </div>
    <div className="infoPage__social--inner infoPage__social--link">
      <a
        className="infoPage__social--link infoPage__social--link-skype"
        href="skype:live%3Aanna.branco_3?call"
      >
        <i className="fab fa-skype icon-call" />
        <p className="infoPage__social-text">{texts.call}</p>
      </a>
      <a
        className="infoPage__social--link infoPage__social--link-skype"
        href="skype:live%3Aanna.branco_3?chat"
      >
        <i className="fas fa-comments icon-chat" />
        <p className="infoPage__social-text">{texts.chat}</p>
      </a>
    </div>
  </div>
);

Social.propTypes = {
  texts: infoPageTextPropType.isRequired
};

export default Social;
