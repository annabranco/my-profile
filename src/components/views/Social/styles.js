import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorWhite,
  colorBlueDark,
  colorBlack,
  fontTitleAlt,
  colorGreenBright,
  colorBlueNormal
} from '../../../styles/theme';

export const SocialWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border-radius: 30px;
  background-image: linear-gradient(
    ${rgba(colorGreenBright, 0.1)},
    ${rgba(colorWhite, 0.2)}
  );
  height: 400px;
  padding: 10px 0 30px;

  @media all and (min-width: 500px) {
    height: 500px;
    padding: 10px 0 80px;
  }
`;

export const SocialBlock = styled.div`
  margin: 5px 0;
  width: 200px;

  @media all and (min-width: 500px) {
    margin: 15px 0;
  }
`;

export const SocialBlockSkype = styled(SocialBlock)`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const SocialLink = styled.a`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  text-decoration: none;

  color: ${colorBlueNormal};
  & :hover {
    color: ${colorBlueDark};
  }
`;

export const SocialLinkButton = styled(SocialLink)`
  border: 1px solid ${colorWhite};
  border-radius: 25px 0;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.2)},
    ${rgba(colorGreenBright, 0.5)}
  );
  width: 120px;
  padding: 5px 0;
  text-decoration: none;
  box-shadow: 0 4px 5px 0 ${rgba(colorBlack, 0.4)},
    inset 0 1px 5px 1px ${rgba(colorWhite, 0.7)};

  @media all and (min-width: 321px) {
    width: 150px;
  }

  @media all and (min-width: 500px) {
    width: 200px;
  }

  &:hover {
    transform: translate(0, 1px);
    text-decoration: none;
    box-shadow: inset 0 2px 5px 1px ${rgba(colorBlack, 0.4)};
  }
`;

export const SocialLinkSkypeButton = styled(SocialLinkButton)`
  width: 75px;
  flex-direction: column;
  width: 100%;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }

  @media all and (min-width: 500px) {
    width: 50%;
  }

  &:first-of-type {
    border-radius: 25px 0 0 0;
  }

  &:last-of-type {
    border-radius: 0 0 25px 0;
  }
`;

export const SocialIcon = styled.div`
  display: inline;
  vertical-align: middle;
  text-shadow: 0 0 1px ${colorBlack};
  font-size: 2rem;
`;

export const SocialIconSkype = styled(SocialIcon)`
  & :hover {
    color: ${colorBlueDark};
  }
`;

export const SocialName = styled.p`
  margin: 4px 0 0;
  font-family: ${fontTitleAlt};
  font-size: 1rem;
  color: ${colorBlack};


  ${SocialLinkButton} & {
    margin: 4px 0 0 25px;
  }

  ${SocialLinkSkypeButton} & {
    margin: 4px 0 0 0;
  }

  @media all and (min-width: 321px) {
    margin: 4px 0 0 15px;
  }

  @media all and (min-width: 500px) {
    margin: 4px 0 0 0;
    font-size: 1.3rem;

    /* ${SocialLinkButton} &: {
      color: red;
      margin: 4px 0 0 5px;
    } */
    /* .infoPage__social--link-mail &,
    .infoPage__social--link-skype & {
      margin: 4px 25px 0;
    } */
  }

  /* .infoPage__social--link-mail &,
  .infoPage__social--link-skype & {
  } */
`;
