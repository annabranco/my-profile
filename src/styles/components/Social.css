.hero__social {
  border-radius: 30px;
  background-image: linear-gradient(rgba(19, 175, 7, 0.1), rgba(255, 255, 255, 0.2));
  height: 500px;
  display: flex;
  padding: 10px 0 80px;
  flex-direction: column;
  align-items: center;
  justify-content: space-around; }

.hero__social--inner {
  margin: 15px 0; }

.hero__social--link,
.hero__social--link-mail,
.hero__social--link-skype {
  display: flex;
  align-items: center;
  justify-content: center; }

.hero__social--link {
  border: 1px solid white;
  border-radius: 25px 0;
  background-image: linear-gradient(rgba(255, 255, 255, 0.2), rgba(47, 195, 74, 0.5));
  width: 200px;
  padding: 5px 0;
  text-decoration: none;
  box-shadow: 0 4px 5px 0 rgba(17, 17, 17, 0.4), inset 0 1px 5px 1px rgba(255, 255, 255, 0.7); }
  .hero__social--link:hover {
    text-decoration: none;
    transform: translate(0, 1px);
    box-shadow: inset 0 2px 5px 1px rgba(0, 0, 0, 0.4); }

.hero__social--inner + .hero__social--link {
  border: 0;
  border-radius: 0;
  background-image: none;
  width: 200px;
  padding: 0;
  text-decoration: none;
  box-shadow: none; }
  .hero__social--inner + .hero__social--link:hover {
    text-decoration: none;
    transform: none;
    box-shadow: none; }

.hero__social--link-mail,
.hero__social--link-skype {
  flex-direction: column;
  width: 100%;
  text-decoration: none; }
  .hero__social--link-mail:hover,
  .hero__social--link-skype:hover {
    text-decoration: none; }

.hero__social--link-skype {
  width: 50%; }
  .hero__social--link-skype:first-of-type {
    border-radius: 25px 0 0 0; }
  .hero__social--link-skype:last-of-type {
    border-radius: 0 0 25px 0; }

.icon-call,
.icon-chat,
.icon-github,
.icon-linkedin,
.icon-mail,
.icon-twitter {
  display: inline;
  vertical-align: middle;
  font-size: 2rem;
  text-shadow: 0 0 1px black; }

.hero__social--link-skype:hover .icon-call, .hero__social--link-skype:hover
.icon-chat {
  color: rgba(5, 65, 135, 0.9); }

.icon-github,
.icon-linkedin,
.icon-mail,
.icon-twitter {
  display: inline;
  vertical-align: middle;
  font-size: 2rem;
  text-shadow: 0 0 1px black; }
  .hero__social--link:hover .icon-github, .hero__social--link:hover
  .icon-linkedin, .hero__social--link:hover
  .icon-mail, .hero__social--link:hover
  .icon-twitter {
    color: rgba(5, 65, 135, 0.9); }

.hero__social-text {
  margin: 4px 0 0 25px;
  font-family: "Acme", sans-serif;
  font-size: 1.3rem;
  color: black; }
  .hero__social--link-mail .hero__social-text,
  .hero__social--link-skype .hero__social-text {
    margin: 4px 25px 0; }
