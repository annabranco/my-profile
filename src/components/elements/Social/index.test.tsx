import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import Social from '.';

// TODO [07-May -21]: Define real test suits (Anna Branco)

const mockTexts = {
  aditional: '',
  advise: '',
  call: 'Call',
  chat: 'Chat',
  job: '',
  thankyou: ''
};

describe('< Social >', () => {
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    wrapper = shallow(<Social texts={mockTexts} />);
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
