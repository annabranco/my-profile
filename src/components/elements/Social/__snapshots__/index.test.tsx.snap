// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Social > matches the snapshot 1`] = `
<Social Area>
  <Social Item>
    <Link
      href="mailto:anya.branco@icloud.com"
    >
      <Icon
        className="far fa-envelope"
      />
      <Text>
        anya.branco@icloud.com
      </Text>
    </Link>
  </Social Item>
  <Social Item>
    <Link Button
      href="https://github.com/annabranco"
      rel="noopener noreferrer"
      target="_Blank"
    >
      <Icon
        className="fab fa-github-alt"
      />
      <Text>
        Github
      </Text>
    </Link Button>
  </Social Item>
  <Social Item>
    <Link Button
      href="https://www.linkedin.com/in/annabranco/"
      rel="noopener noreferrer"
      target="_Blank"
    >
      <Icon
        className="fab fa-linkedin-in"
      />
      <Text>
        LinkedIn
      </Text>
    </Link Button>
  </Social Item>
  <Social Item>
    <Link Button
      href="https://twitter.com/AnyaBranco"
      rel="noopener noreferrer"
      target="_Blank"
    >
      <Icon
        className="fab fa-twitter"
      />
      <Text>
        Twitter
      </Text>
    </Link Button>
  </Social Item>
  <Social Item>
    <Link Button 
      href="skype:live%3Aanna.branco_3?call"
    >
      <Icon
        className="fab fa-skype"
      />
      <Text>
        Call
      </Text>
    </Link Button >
    <Link Button 
      href="skype:live%3Aanna.branco_3?chat"
    >
       
      <Icon
        className="fas fa-comments"
      />
      <Text>
        Chat
      </Text>
    </Link Button >
  </Social Item>
</Social Area>
`;
